### PR TITLE
Exam agent deployment

### PR DESCRIPTION
**Pull request by QcFe**
# Description

This PR includes helm chart deployment information for the CrownLabs Exams Agent

### CODE REVIEW

File: operators/deploy/exam-agent/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+kind: Ingress
+metadata:
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt-production
+    nginx.ingress.kubernetes.io/cors-allow-credentials: "true"
+    nginx.ingress.kubernetes.io/cors-allow-methods: PUT, GET, DELETE
+    nginx.ingress.kubernetes.io/enable-cors: "true"
+    nginx.ingress.kubernetes.io/proxy-connect-timeout: "600"
+    nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
+    nginx.ingress.kubernetes.io/proxy-send-timeout: "600"
```
Comment (giorio94) [inline operators/deploy/exam-agent/templates/ingress.yaml:11]: Are those annotations required?

File: operators/deploy/exam-agent/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  name: {{ include "exam-agent.fullname" . }}
+spec:
+  ingressClassName: nginx-external
```
Comment (giorio94) [inline operators/deploy/exam-agent/templates/ingress.yaml:14]: Is this specified in the other manifests?

File: operators/deploy/exam-agent/templates/rolebindings.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  labels:
+    {{- include "exam-agent.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: crownlabs-view-instances
+subjects:
+  - kind: ServiceAccount
+    name: {{ include "exam-agent.fullname" . }}
+    namespace: {{ .Release.Namespace }}
```
Comment (giorio94) [inline operators/deploy/exam-agent/templates/rolebindings.yaml:47]: This should not be necessary, as already implied by manage-instances

File: deploy/crownlabs/Chart.lock [hunk 4]
```diff
@@ -23,6 +23,8 @@ dependencies:
 - name: delete-stale-instances
   repository: file://../../operators/deploy/delete-stale-instances
   version: 0.1.0
+- name: exam-agent
+  repository: file://../../operators/deploy/exam-agent
```
Comment (giorio94) [inline deploy/crownlabs/Chart.lock:27]: Is it correct that the version is missing?

**Review by giorio94 — APPROVED** [review]

File: operators/deploy/exam-agent/templates/ingress.yaml (hunk 2 above)
Comment (QcFe) [reply operators/deploy/exam-agent/templates/ingress.yaml:14]: Don't know where this came from

File: operators/deploy/exam-agent/templates/ingress.yaml (hunk 1 above)
Comment (giorio94) [reply operators/deploy/exam-agent/templates/ingress.yaml]: There is no guarantee that those on the api-server are needed :-)

Comment (QcFe) [reply operators/deploy/exam-agent/templates/ingress.yaml]: put an ingressAnnotations on `values.yaml`

**Comment by QcFe** [comment]
@giorio94 feel free to merge if it's alright

**Comment by giorio94** [comment]
> @giorio94 feel free to merge if it's alright

A check seems to be broken

**Pull request merged**